GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -375,24 +379,24 @@ public virtual void OnIndexFileChange()
                     { TracingConstants.MessageKey.WarningMessage, "Index modified without git holding GVFS lock" },
                 };
 
-                this.context.Tracer.RelatedEvent(EventLevel.Warning, $"{nameof(this.OnIndexFileChange)}_NoLock", metadata);
+                this.context.Tracer.RelatedEvent(EventLevel.Warning, $"{nameof(this.CheckLockWhenIndexChanged)}_NoLock", metadata);
             }
-            else if (this.GitCommandLeavesProjectionUnchanged(gitCommand))
-            {
-                if (this.GitCommandRequiresModifiedPathValidationAfterIndexChange(gitCommand))
-                {
-                    this.GitIndexProjection.InvalidateModifiedFiles();
-                    this.backgroundFileSystemTaskRunner.Enqueue(FileSystemTask.OnIndexWriteRequiringModifiedPathsValidation());
-                }
+        }
 
-                this.InvalidateGitStatusCache();
-            }
-            else
+        public void OnIndexFileChanged(bool workingDirectoryUpdated, bool onlyIndexUpdated)

```

Comment:
Maybe rename this to `OnIndexFileChangedHook` to make it clearer that this method is driven by the git hook and not by file notifications.